x-pack/plugins/security_solution/public/resolver/store/ui/selectors.ts

```diff
@@ -31,20 +35,129 @@ export const selectedNode = createSelector(
+    if (value === null) {
+      // Equivalent to `null`
+      return defaultParameters();
+    }
+    const decodedValue: unknown = decode(value);
+    if (isPanelViewAndParameters(decodedValue)) {
+      return decodedValue;
     }
-    return locationSearchModel.breadcrumbParameters(locationSearch, resolverComponentInstanceID);
+    return defaultParameters();
   }
 );
+
+/**
+ * Return a relative href (which includes just the 'search' part) that contains an encoded version of `params `.
+ * All other values in the 'search' will be kept.
+ * Use this to get an `href` for an anchor tag.
+ */
+export const relativeHref: (
+  state: ResolverUIState
+) => (params: PanelViewAndParameters) => string | undefined = createSelector(
+  (state: ResolverUIState) => state.locationSearch,
+  (state: ResolverUIState) => state.resolverComponentInstanceID,
+  (locationSearch, resolverComponentInstanceID) => {
+    return (params: PanelViewAndParameters) => {
+      /**
+       * This is only possible before the first `'appReceivedNewExternalProperties'` action is fired.
+       */
+      if (locationSearch === undefined || resolverComponentInstanceID === undefined) {
+        return undefined;
+      }
+      const urlSearchParams = new URLSearchParams(locationSearch);
+      const value = encode(params);
+      urlSearchParams.set(parameterName(resolverComponentInstanceID), value);
+      return `?${urlSearchParams.toString()}`;
+    };
+  }
+);
+
+/**
+ * Returns a map of ecs category name to urls for use in panel navigation.
+ * @deprecated
+ */
+export const relatedEventsRelativeHrefs: (
+  state: ResolverUIState
+) => (
+  categories: Record<string, number> | undefined,
+  nodeID: string
+) => Map<string, string | undefined> = createSelector(relativeHref, (relativeHref) => {
+  return (categories: Record<string, number> | undefined, nodeID: string) => {
+    const hrefsByCategory = new Map<string, string | undefined>();
+    if (categories !== undefined) {
+      Object.keys(categories).map((category) => {
+        const categoryPanelParams: PanelViewAndParameters = {
+          panelView: 'nodeEventsOfType',
+          panelParameters: {
+            nodeID,
+            eventType: category,
+          },
+        };
+        hrefsByCategory.set(category, relativeHref(categoryPanelParams));
+        return category;
+      });
+    }
+    return hrefsByCategory;
+  };
+});
+
+/**
+ * Returns a map of event entity ids to urls for use in navigation.
+ * @deprecated
+ */
+export const relatedEventDetailHrefs: (
+  state: ResolverUIState
+) => (
+  category: string,
+  nodeID: string,
+  events: ResolverEvent[]
+) => Map<string, string | undefined> = createSelector(relativeHref, (relativeHref) => {
+  return (category: string, nodeID: string, events: ResolverEvent[]) => {
+    const hrefsByEntityID = new Map<string, string | undefined>();
+    events.map((event) => {
+      const entityID = String(eventId(event));
```

Comment:
Using `entityID` might be a little confusing here. Is this supposed to contain `process.entity_id` or `event.id`?

Reply:
what have we done

Reply:
The variable name is wrong, it is supposed to be `eventId`

Reply:
Fixed in next PR

Reply:
I think this was me being lazy to avoid the no-shadow lint rule, will fix!